src/nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py

```diff
@@ -45,17 +48,48 @@ def extract_electrode_metadata_with_pyintan(file_path):
     return electrodes_metadata
 
 
-class IntanRecordingInterface(BaseRecordingExtractorInterface):
-    """Primary data interface class for converting a IntanRecordingExtractor."""
+def extract_electrode_metadata(recording_extractor: BaseRecording):
+
+    channel_name_array = recording_extractor.get_property("channel_name")
+
+    group_names = [channel.split("-")[0] for channel in channel_name_array]
+    unique_group_names = set(group_names)
+    group_electrode_numbers = [int(channel.split("-")[1]) for channel in channel_name_array]
+    custom_names = list()
+
+    electrodes_metadata = dict(
+        group_names=group_names,
+        unique_group_names=unique_group_names,
+        group_electrode_numbers=group_electrode_numbers,
+        custom_names=custom_names,
+    )
+
+    return electrodes_metadata
 
-    RX = se.IntanRecordingExtractor
 
-    def __init__(self, file_path: FilePathType, verbose: bool = True):
-        assert HAVE_PYINTAN, INSTALL_MESSAGE
-        super().__init__(file_path=file_path, verbose=verbose)
+class IntanRecordingInterface(BaseRecordingExtractorInterface):
+    """Primary data interface class for converting a IntanRecordingExtractor."""
 
-        electrodes_metadata = extract_electrode_metadata_with_pyintan(file_path)
-        self.recording_extractor = OldToNewRecording(oldapi_recording_extractor=self.recording_extractor)
+    RX = IntanRecordingExtractor
+
+    def __init__(
+        self,
+        file_path: FilePathType,
+        stream_id: Optional[str] = "0",
+        spikeextractors_backend: Optional[bool] = False,
+        verbose: Optional[bool] = True,
```

Comment:
Remove the boolean `Optional`, as Ben explained in the other PR. Can `stream_id` be `None`?

Reply:
No,t it can't be None for this recorder. Removing it.